Draft guide for OCI Kubernetes

n0tl3ss opened the pull request:
- Added guide for OCI Kubernetes
- Small changes to Kubernetes base guide
- Added `external-template` and `common-template`, Uses external resources as templates you can provide argument that will be replaced in text.

For an example:

`external-template:micronaut-k8s-oci/k8s-microservice.adoc[arg0=api]`

will include `micronaut-k8s-oci/k8s-microservice.adoc` and substitute every appearance of `{0}` with `api`.

PDFs:

[Using Oracle Cloud Container Engine for Kubernetes (OKE) and the Micronaut Framework.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10189076/Using.Oracle.Cloud.Container.Engine.for.Kubernetes.OKE.and.the.Micronaut.Framework.pdf)

[Kubernetes and the Micronaut Framework.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10189088/Kubernetes.and.the.Micronaut.Framework.pdf)

timyates committed:
Switch to use Serde for the S3 Event guide (#1138)

timyates committed:
Switch to use Serde for the Discovery eureka guide (#1134)

timyates committed:
Switch to use Serde for the Oauth client credentials auth0 guide (#1133)

timyates committed:
Switch to use Serde for the Discovery consul guide (#1135)

timyates committed:
Use the correct version for the micronaut-data processor with SpringBoot (#1164)

We were using the version of Micronaut, which is incorrect

sbodvanski committed:
Guide: Connect a Micronaut JMS Application to an AWS SQS Queue (#1160)

Max supported Java version by AWS SDK 1.x is 16 (https://github.com/aws/aws-sdk-java#maintenance-and-support-for-java-versions)

sdelamo committed:
add AWS SQS twitter card

wetted committed:
Remove references to postgres and use mysql consistently for guide (#1166)

closes #1161

timyates committed:
Switch to use Serde for the Mybatis data access guide (#1139)

* Switch to use Serde for the Mybatis data access guide

* Remove Creator, fix Kotlin warning

n0tl3ss committed:
Fixes missing section in trace guide (#1173)

n0tl3ss committed:
Fixes wrong callouts to WarehouseController endpoints (#1174)

n0tl3ss committed:
Draft guide for OCI Kubernetes

sachin-pikle commented:
Hi @n0tl3ss,

Some feedback:
1. Kubernetes and the Micronaut Framework.pdf: Correct the secret name in the footnote 4 below the auth.yml in section 5.
<img width="531" alt="image" src="https://user-images.githubusercontent.com/29235281/207334177-fd56202a-c072-4a3f-91b4-9009aa91da70.png">

2. Both the guides: Instead of using "kubectl create -f ...yml", it's better to use "kubectl apply -f ...yml"
_(Recommended by K8s https://kubernetes.io/docs/reference/kubectl/cheatsheet/#kubectl-apply)_

3. Kubernetes and the Micronaut Framework.pdf: Works with micronaut-kubernetes version 3.4.1-SNAPSHOT.

4. Another suggestion would be to have a separate k8s namespace to create the resources used in the guide. Cleaning up is as easy as deleting the namespace.

n0tl3ss committed:
Merge branch 'feature/k8s-guide' into feature/k8s-guide-oci

n0tl3ss committed:
Apply suggestions from code review

Co-authored-by: Bernard Horan <bernard.horan@oracle.com>

sdelamo committed:
Update guides to 3.7.5 (#1179)

n0tl3ss committed:
code review improvements add native image support

n0tl3ss committed:
code review improvements add native image support

n0tl3ss commented:
New versions: 

[Kubernetes and the Micronaut Framework Maven.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10270554/Kubernetes.and.the.Micronaut.Framework.Maven.pdf)
[Kubernetes and the Micronaut Framework Gradle.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10270555/Kubernetes.and.the.Micronaut.Framework.Gradle.pdf)
[Using Oracle Cloud Container Engine for Kubernetes (OKE) and the Micronaut Framework Maven.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10270556/Using.Oracle.Cloud.Container.Engine.for.Kubernetes.OKE.and.the.Micronaut.Framework.Maven.pdf)
[Using Oracle Cloud Container Engine for Kubernetes (OKE) and the Micronaut Framework Gradle.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10270558/Using.Oracle.Cloud.Container.Engine.for.Kubernetes.OKE.and.the.Micronaut.Framework.Gradle.pdf)

n0tl3ss committed:
code review improvements

n0tl3ss committed:
code review improvements

n0tl3ss commented:
New version of guides:

[Using Oracle Cloud Container Engine for Kubernetes (OKE) and the Micronaut Framework Gradle.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10295513/Using.Oracle.Cloud.Container.Engine.for.Kubernetes.OKE.and.the.Micronaut.Framework.Gradle.pdf)

[Kubernetes and the Micronaut Framework Gradle.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10295510/Kubernetes.and.the.Micronaut.Framework.Gradle.pdf)

[Using Oracle Cloud Container Engine for Kubernetes (OKE) and the Micronaut Framework Maven.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10295511/Using.Oracle.Cloud.Container.Engine.for.Kubernetes.OKE.and.the.Micronaut.Framework.Maven.pdf)

[Kubernetes and the Micronaut Framework Maven.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10295507/Kubernetes.and.the.Micronaut.Framework.Maven.pdf)

timyates committed:
Build Guides with Micronaut 3.8.0 (#1157)

* Build Guides with Micronaut 3.8.0

This PR runs the guides with Micronaut 3.8.0-SNAPSHOT as a pre-release check

Removes the local localstack feature as this now exists in 3.8.0

* Add snapshot repo for snapshot versions of Micronaut

* Look up the version for the data processor

* version 3.8.0

Co-authored-by: Sergio del Amo <sergio.delamo@softamo.com>

timyates committed:
Remove ambiguity for Kotlin (#1182)

It couldn't work out which mathod to call as the closure had an implicit return value.

I'm not sure why this has only recently appeared...  Maybe a change to micrometer

timyates committed:
Fix inputs for guides with a base (#1183)

If we have a guide that declares a base project, ie:

https://github.com/micronaut-projects/micronaut-guides/blob/dd911994e0c54c068602389dbcbe08eac8bdf045/guides/micronaut-metrics-aws/metadata.json#L2

Then we were not taking the code from micronaut-metrics into account for the cache-key, so changes were not picked up when building micronaut-metrics-aws

This change adds the base directory as an input if it was set

sdelamo committed:
remove empty constructor (#1167)

burtbeckwith committed:
misc fixes and cleanup (#1186)

graemerocher committed:
Add R2DBC guide (#965)

sdelamo committed:
copy: change GraalVM version (#1187)

sdelamo committed:
change publication to 2023-01-04

sdelamo committed:
fix: category should be categories

timyates committed:
disable spring-boot-micronaut-data guide (#1185)

sdelamo committed:
Micronaut Framework 3.8.1 (#1195)

sdelamo committed:
Kotlin and Groovy versions of Micronaut Dependency Injection types (#1227)

Close #1082

n0tl3ss committed:
Add K8s AWS guide

n0tl3ss commented:
Added AWS guide:
[Using Amazon Elastic Kubernetes Service (EKS) and the Micronaut Framework.pdf](https://github.com/micronaut-projects/micronaut-guides/files/10435004/Using.Amazon.Elastic.Kubernetes.Service.EKS.and.the.Micronaut.Framework.pdf)

n0tl3ss committed:
Add K8s AWS guide

sdelamo committed:
Kotlin version of JAX-RS guide (#1225)

Close #1221

n0tl3ss committed:
code review improvements

sdelamo committed:
Micronaut Framework 3.8.2 (#1229)

alina-yur committed:
fix java version typo (#1234)

sdelamo committed:
Mock Github Server to avoid rate limiting in HTTP Client Guide (#1236)

* build: Micronaut Framework 3.8.4

sdelamo committed:
X-API-KEY guide (#1237)

sdelamo committed:
add api key guide twitter card

sdelamo committed:
CORS guide (#1238)

sdelamo committed:
docker image guide (#1239)

sdelamo committed:
executable Jar guide to distribution category (#1240)

- enable every language. Use distributable-base

sdelamo committed:
create the http client lazyly (#1243)

sdelamo committed:
add lateinit and @field callouts (#1228)

sdelamo committed:
Custom constraint guide (#1244)

* custom constraint

* Update guides/micronaut-custom-validation-annotation/java/src/test/java/example/micronaut/ContactTest.java

sdelamo committed:
twitter card

hartraft committed:
Fix curl command (#1248)

When copying the curl command, each new line is executed rather than waiting for the last command.

sdelamo committed:
build: Micronaut Framework 3.8.5(#1249)

sdelamo committed:
Guide: Using IntelliJ IDEA to develop Micronaut applications (#1241)

sdelamo committed:
2023 copyright

sdelamo committed:
Micronaut Framework 3.8.6 (#1251)

sdelamo committed:
build: Micronaut Framework 3.8.7 (#1252)

sdelamo committed:
add awaitility version (#1253)

Close: https://github.com/micronaut-projects/micronaut-guides/issues/1074

phamtienlong19 committed:
Fixes for MN Scope Types guide, mostly imports and annotations to ensure that the Tests work as intended. (#1254)

phamtienlong19 committed:
[Error-handling Guide] Added missing dependency and a closing bracket for the BookController class. (#1260)

sdelamo committed:
Micronaut Framework 3.8.8 (#1261)

sdelamo committed:
move create your first graalvm guide to distribution category (#1256)

phamtienlong19 committed:
Security Session Guide: Removed the extra dependency that could cause tests to fail (#1263)

sdelamo committed:
CRaC Cloud Run guide (#1259)

Close #1257

sdelamo committed:
Keycloak Guide (#586)

sdelamo committed:
add back common-google-cloud-platform.adoc snippet

accidentally deleted via https://github.com/micronaut-projects/micronaut-guides/pull/1259

sdelamo committed:
add image back

sdelamo committed:
use categories

sdelamo committed:
Micronaut Framework 3.8.9 (#1264)

phamtienlong19 committed:
[Database JDBC Guides for AWS, Azure, and GCP] Fixed the environment variable names to be compatible with the sample JDBC application, and added explanations similar to the original JDBC guide. (#1270)

sdelamo committed:
build: Micronaut Framework 3.9.1 (#1267)

sdelamo committed:
build: Micronaut Build Plugin to 5.4.9 (#1273)

sdelamo committed:
3.9.2 (#1280)

sdelamo committed:
Merge branch 'main' into k8s-guide
